backend/src/packages/users/user.entity.ts

```diff
@@ -2,49 +2,57 @@ import { type Entity } from '#libs/types/types.js';
 
 class UserEntity implements Entity {
   private id: number | null;
-
+  private createdAt: Date | null;
+  private updatedAt: Date | null;
   private email: string;
-
   private passwordHash: string;
-
   private passwordSalt: string;
-
   private constructor({
     id,
     email,
     passwordHash,
     passwordSalt,
+    createdAt,
+    updatedAt,
   }: {
     id: number | null;
     email: string;
     passwordHash: string;
     passwordSalt: string;
+    createdAt: Date | string | null;
+    updatedAt: Date | string | null;
```

Comment:
Hey) I known that there were a long discussion, but let's return back to the date approach here
```suggestion
    createdAt: Date | null;
    updatedAt: Date | null;
```
and map them when you call methods, as you did before, but with all properties, like this:
```
UserEntity.initialize({
  id: payload.id
  email: payload.email,
  fullName: payload.fullName,
  createdAt: new Date(payload.createdAt),
  updatedAt: new Date(payload.createdAt)
}),
```
sorry for that 🙏

Reply:
It's ok, it happens. Thank you for clarifications))